internal/wsjs/wsjs.go

```diff
@@ -119,23 +131,29 @@ func (c *WebSocket) OnMessage(fn func(m MessageEvent)) {
 	})
 }
 
-func (c *WebSocket) OnOpen(fn func(e js.Value)) {
-	c.addEventListener("open", fn)
+// OnOpen registers a function to be called when the websocket is opened.
+func (c WebSocket) OnOpen(fn func(e js.Value)) (remove func()) {
+	return c.addEventListener("open", fn)
 }
 
-func (c *WebSocket) Close(code int, reason string) (err error) {
+// Close closes the WebSocket with the given code and reason.
+func (c WebSocket) Close(code int, reason string) (err error) {
 	defer handleJSError(&err, nil)
 	c.v.Call("close", code, reason)
 	return err
 }
 
-func (c *WebSocket) SendText(v string) (err error) {
+// SendText sends the given string as a text message
+// on the WebSocket.
+func (c WebSocket) SendText(v string) (err error) {
 	defer handleJSError(&err, nil)
 	c.v.Call("send", v)
```

Comment:
Do we have good test coverage for the `SendText` method? I've had a lot of trouble with things like this in the past because JavaScript does wonky and unexpected things with text encoding. The string you are trying to send might not end up being the same as what is received on the other end. It might even be better to just remove this method.

Reply:
We have good test coverage here. See https://github.com/nhooyr/websocket/blob/ef5a88d1828ab3da40b07b9faacf627b0421b329/websocket_js_test.go#L39

Reply:
@nhooyr it looks like [`randString` only includes hex characters](https://github.com/nhooyr/websocket/blob/7b4bd30d74307696a93536d863763f627e459c84/assert_test.go#L102)? You should probably test with unicode characters, in particular ones with higher code points. There are some differences in how these are encoded in Go and JavaScript.

Reply:
That's fair. I'll adjust it.

Reply:
Fixed in #148